### ISSUE TITLE
InfoNCE implementation bug

### ISSUE THREAD

**Issue by isaaccorley** [YihengZhang-CV/SeCo-Sequence-Contrastive-Learning#3]
It seems as though your `NCESoftmaxLoss` implementation is simply learning that everything is a negative sample because your ground truth  are all zeros `label = torch.zeros([x.shape[0]]).long().to(x.device)`.

https://github.com/YihengZhang-CV/SeCo-Sequence-Contrastive-Learning/blob/f0b1b48731de808694e57da348e366df57dcd8c7/seco/Contrast.py#L6-L14
